src/admin/apiAdminCommunication.ts

```diff
@@ -0,0 +1,50 @@
+import 'whatwg-fetch';
+
+const currentView = document.getElementById('app').getAttribute('name');
+
+export const sendAuthRequest = ((data: any): Promise<any> => {
+	const apiUrl = `/admin/login/`;
+	return fetch(apiUrl, {
+		method: 'POST',
+		headers: { 'Content-Type': 'application/json' },
+		body: JSON.stringify(data),
+		credentials: 'include',
+	})
+		.then(authCheck)
+		.then(json)
+		.then(function (json) {
+			return json;
+		}).catch(function (error) {
+			console.log('request failed', error);
+		});
+});
+
+function authCheck(response) {
+	const currentPath = window.location.pathname;
+
+	if (currentPath === '/admin/login') {
```

Comment:
move it to separate function, outside of authCheck(if we will use this func to check all our requests)
```
if (currentPath === '/admin/login') {
 		if (response.status !== 200) {
 			return response;
 		}
 		window.location.pathname = 'admin/testpage';
 		return response;
 }
```

I think authCheck should be something like this - 

```
function authCheck(response) {
if (response.status === 401) {
  UIActions.showLoginModal();
 }
return response;
}
```